field/FieldCreate.F90

```diff
@@ -116,6 +116,9 @@ subroutine field_empty_complete( field, &
            ungriddedUBound=bounds%upper, &
            _RC)
 
+      ! Initialize field to zero
+      call ESMF_FieldFill(field, dataFillScheme="const", const1=0.d0, _RC)
```

Comment:
As per the discussion yesterday, I think that setting the initial value to 0 will tend to hide bugs.    In the use case you mentioned, the 2nd field should not have been referenced, and indeed should not even exist.

If you need this as a short term workaround, then fine.  But please then come back to either delete or initialize to some absurd value.